### PR TITLE
Add, remove, reorder, and group by metadata tracks on the heatmap

### PR DESCRIPTION
**Pull request by marcellp**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-417
https://biomage.atlassian.net/browse/BIOMAGE-419

#### Link to staging deployment URL
https://ui-metadata-track-settings.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
N/A

#### Anything else the reviewers should know about the changes here
N/A

# Changes
### Code changes
Added a transfer-style component to the heatmap settings menu to be able to quickly add, remove, reorder, and group by metadata tracks.

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock
- [x] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [x] Relevant Github READMEs updated
- [x] Relevant wiki pages created/updated

### Approvers
- [x] Approved by a member of the core engineering team
- [x] (UX changes) Approved by @vickymorrison

### Just before merging:
- [x] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [x] Photo of a cute animal attached to this PR
image

### CODE REVIEW

**Comment by marcellp** [comment]
Staging deployed at https://ui-metadata-track-settings.scp-staging.biomage.net/ @ivababukova @vickymorrison

**Review by vickymorrison — APPROVED** [review]

File: assets/antd-custom.less [hunk 1]
```diff
@@ -1,2 +0,0 @@
-@color-primary: #008DA6;
-@link-color: #00A3BF;
```
Comment (ivababukova) [inline assets/antd-custom.less]: why is this deleted now?

File: src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx [hunk 2]
```diff
@@ -37,9 +37,14 @@ const HeatmapPlot = (props) => {
   const properties = useSelector((state) => state.cellSets.properties);
   const hidden = useSelector((state) => state.cellSets.hidden);
 
+  const selectedTracks = useSelector((state) => state.cellInfo.selectedTracks);
+  const groupedTrack = useSelector((state) => state.cellInfo.groupedTrack);
```
Comment (ivababukova) [inline src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx]: Have you thought about whether this approach still keeps our redux store extensible -- for instance, will values for `selectedTracks` and `groupedTrack` apply to all components similarly to how the `cellId` in the `cellInfo` redux store applies to everything? Or would values of `selectedTracks` and `groupedTrack` be different for different views of the dataset?
Just wondering whether `cellInfo` is the right place in the store to be placing these.

File: src/__test__/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.test.jsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+jest.mock('../../../../../../../pages/experiments/[experimentId]/data-exploration/components/heatmap/VegaHeatmap');
+
+const mockStore = configureMockStore([thunk]);
+configure({ adapter: new Adapter() });
+
+let component;
+
+const initialState = {
+  cellSets: {
+    // cellSets', 'metadataCategorical
```
Comment (ivababukova) [inline src/__test__/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.test.jsx]: remove this line

File: src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Radio } from 'antd';
+import { updatePlotConfig } from '../../../../../../redux/actions/componentConfig';
+
+const HeatmapGroupBySettings = () => {
+  const dispatch = useDispatch();
+  const groupedTrack = useSelector((state) => state.componentConfig.interactiveHeatmap.config.groupedTrack);
```
Comment (ivababukova) [inline src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.jsx:8]: nice idea to put it in the plot config, works out better than I originally imagined. Will that get saved in dynamo similarly to the plot config for plots in plots and tables?

File: src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx [hunk 5]
```diff
@@ -37,13 +38,26 @@ const HeatmapPlot = (props) => {
...
   const setDataDebounce = useCallback(_.debounce((data) => {
     setVegaData(data);
   }, 1500, { leading: true }), []);
 
+  useEffect(() => {
+    if (!_.isEmpty(heatmapSettings)) {
+      return;
+    }
+
+    dispatch(loadComponentConfig(experimentId, 'interactiveHeatmap', 'interactiveHeatmap'));
```
Comment (ivababukova) [inline src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx:57]: why are you supplying `interactiveHeatmap` twice?

File: src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx [hunk 6]
```diff
@@ -37,13 +38,26 @@ const HeatmapPlot = (props) => {
   const properties = useSelector((state) => state.cellSets.properties);
   const hidden = useSelector((state) => state.cellSets.hidden);
 
+  const heatmapSettings = useSelector((state) => state.componentConfig.interactiveHeatmap?.config) || {};
```
Comment (ivababukova) [inline src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx:40]: We should have `interactiveHeatmap` assigned to a variable - it is not great to have it hardcoded like this. This `interactiveHeatmap` sounds to me more like component type, but we already have `const componentType = 'Heatmap';` on line 26. I am thinking that these two have to be the same thing actually -- how about we make `const componentType = 'InteractiveHeatmap';` and then we use `componentType` instead of this hardcoded value?
Another way to do it would be to create a variable called `componentUuid` and make it equal to `interactiveHeatmap`.

File: src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.jsx (hunk 4 above)
Comment (marcellp) [reply src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.jsx:8]: No, it wouldn't. I don't think that's a good idea for a component that a user would modify a lot. The save action never gets called.

File: src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx (hunk 6 above)
Comment (marcellp) [reply src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx:40]: yeah, i was thinking of a way to do this properly, and considered basically what you described. will edit.

File: assets/antd-custom.less (hunk 1 above)
Comment (marcellp) [reply assets/antd-custom.less]: i was trying something out but it's better to have it back. reverted.

**Review by ivababukova — APPROVED** [review]
